client/deviceauth/client_deviceauth.go

```diff
@@ -60,6 +57,13 @@ type StatusReq struct {
 	Status   string `json:"status"`
 }
 
+type PreAuthReq struct {
+	DeviceId  string `json:"device_id" valid:"required" bson:"device_id"`
+	AuthSetId string `json:"auth_set_id" valid:"required" bson:"auth_set_id"`
+	IdData    string `json:"id_data" valid:"required" bson:"id_data"`
+	PubKey    string `json:"pubkey" valid:"required" bson:"pubkey"`
```

Comment:
the convention across `devadm` seems to be `key` (also - in the docs for this ep)